vulnerabilities/importers/mozilla.py

```diff
@@ -0,0 +1,208 @@
+from typing import Set, List, Generator
+
+import re
+import asyncio
+from bs4 import BeautifulSoup
+from packageurl import PackageURL
+import requests
+
+import yaml
+from markdown import markdown
+
+from vulnerabilities.data_source import Advisory
+from vulnerabilities.data_source import GitDataSource
+from vulnerabilities.data_source import Reference
+from vulnerabilities.data_source import VulnerabilitySeverity
+from vulnerabilities.severity_systems import scoring_systems
+from vulnerabilities.package_managers import GitHubTagsAPI
+from vulnerabilities.helpers import is_cve
+
+
+REPOSITORY = "mozilla/foundation-security-advisories"
+MFSA_FILENAME_RE = re.compile(r"mfsa(\d{4}-\d{2,3})\.(md|yml)$")
+
+
+class MozillaDataSource(GitDataSource):
+    def __enter__(self):
+        super(MozillaDataSource, self).__enter__()
+
+        if not getattr(self, "_added_files", None):
+            self._added_files, self._updated_files = self.file_changes(
+                recursive=True, subdir="announce"
+            )
+
+        # Do we need this ?
+        # self.version_api = GitHubTagsAPI()
+        # self.set_api()
+
+    def set_api(self):
+        repository = "/".join(self.config.repository_url.split("/")[-2:])
+        asyncio.run(self.version_api.load_api([repository]))
+
+    def added_advisories(self) -> Set[Advisory]:
+        return self._load_advisories(self._added_files)
+
+    def updated_advisories(self) -> Set[Advisory]:
+        return self._load_advisories(self._updated_files)
+
+    def _load_advisories(self, files) -> Set[Advisory]:
+        """
+        Yields list of advisories of batch size
+        """
+        files = [
+            f for f in files if f.endswith(".md") or f.endswith(".yml")
+        ]  # skip irrelevant files
+
+        advisories = []
+        for path in files:
+            for advisory in self._to_advisories(path):
+                advisories.append(advisory)
+                if len(advisories) >= self.batch_size:
```

Comment:
I'm unsure if I should just fetch all the advisories then pass them to `batch_advisories` effectively nullifying the benefit of python generators or should I keep generating them according to batch size and yield them.
I've implemented the later in this case.

Reply:
Use `batch_advisories` for now. It has it's own problems and there's https://github.com/nexB/vulnerablecode/issues/338 for that. The generator thing won't do much, since we are importing like 10-20 MBs of data.  

We'll be able to fix the  https://github.com/nexB/vulnerablecode/issues/338 with some minor changes quite easily, if every importer uses the same thing. I don't really want the batching logic repeated in multiple places.